yt/frontends/amrvac/fields.py

```diff
@@ -33,14 +38,14 @@
 class AMRVACFieldInfo(FieldInfoContainer):
     known_other_fields = (
         # Everything in AMRVAC is normalised to dimensionless units, so set units to ""
-        ("rho", ("", ["density"], r"$\rho$")),
-        ("m1", ("", ["momentum_1"], r"$m_1$")),
-        ("m2", ("", ["momentum_2"], r"$m_2$")),
-        ("m3", ("", ["momentum_3"], r"$m_3$")),
-        ("e", ("", ["energy"], r"$e$")),
-        ("b1", ("", [], r"$B_x$")), # todo: check notation (x, y, z) should only apply to cartesian case
-        ("b2", ("", [], r"$B_y$")),
-        ("b3", ("", [], r"$B_z$"))
+        ("rho", (code_density, ["density"], r"$\rho$")),
+        ("m1", (code_momentum, ["momentum_1"], r"$m_1$")),
+        ("m2", (code_momentum, ["momentum_2"], r"$m_2$")),
+        ("m3", (code_momentum, ["momentum_3"], r"$m_3$")),
+        ("e", (code_energy, ["energy"], r"$e$")),
+        ("b1", (code_magnetic, [], r"$B_1$")),
+        ("b2", (code_magnetic, [], r"$B_2$")),
+        ("b3", (code_magnetic, [], r"$B_3$"))
```

Comment:
this is probably correct (Enzo's frontend does this in a comparable way)
Although I'm not sure we need to define those as variables. Is there a reason why you don't directly put strings here ?